src/test/scala/chiselTests/BulkConnectSpec.scala

```diff
@@ -112,4 +112,15 @@ class BulkConnectSpec extends ChiselPropSpec {
     chirrtl should include("connect out1[0], in1[1]")
     chirrtl should include("connect out1[1], in1[0]")
   }
+
+  property("Chisel should emit FIRRTL bulk connect for \"input\" wires") {
+    class MyBundle extends Bundle {
+      val foo = Input(UInt(8.W))
+    }
+    val chirrtl = ChiselStage.emitCHIRRTL(new Module {
+      val w1, w2 = Wire(new MyBundle)
+      w2 <> w1
+    })
+    chirrtl should include("connect w2, w1")
```

Comment:
💯

Reply:
i never thought that `<>` was very sound for connecting two wires, so this is no worse than before 🤷‍♀️

Reply:
Until recently, I didn't know it even _could_ work for two wires. As I mention in https://github.com/chipsalliance/chisel/issues/4218, it does because of the special logic to sometimes emit FIRRTL bulk connects. So that work which was intended to be purely an optimization and not API visible actually extended the `<>` API in some funky ways.

This PR just makes it slightly more consistent in its funkiness 😇